### PR TITLE
West Midlands | ITP September 2025 | Jonathan Boahene | Sprint 2 | Module Structuring and Testing Data

### PR DESCRIPTION
**Pull request by Abayie**
## Learners, PR Template

Self checklist

- [x ] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x ] My changes meet the requirements of the task
- [x ] I have tested my changes
- [x ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

In this pull request I have added improved solutions and comprehensive tests for Sprint 2 of Module-Structuring-and-Testing-Data.
I refactored functions to handle edge cases and ensure correct formatting, such as converting 24-hour time strings to 12-hour clock format with am/pm.
The code now includes robust test cases for different input scenarios, including midnight, noon, single-digit hours, and times close to midnight and noon.
These changes enhance reliability and maintainability, making the code easier to understand and verify for a wide range of data inputs.

## Questions

No questions at this point

### CODE REVIEW

File: Sprint-2/3-mandatory-implement/3-to-pounds.js [hunk 1]
```diff
@@ -4,3 +4,21 @@
 // You will need to declare a function called toPounds with an appropriately named parameter.
 
 // You should call this function a number of times to check it works for different inputs
+
+function toPounds(penceString) {
+  // Remove trailing 'p' and pad with zeros to ensure at least 3 digits
+  const pence = penceString
+    .substring(0, penceString.length - 1)
```
Comment (jaymes15) [inline Sprint-2/3-mandatory-implement/3-to-pounds.js]: This is clever and handles edge cases very well.

File: Sprint-2/3-mandatory-implement/3-to-pounds.js [hunk 2]
```diff
@@ -4,3 +4,21 @@
...
+    .padStart(3, "0");
+  // Get pounds and pence parts
+  const pounds = pence.substring(0, pence.length - 2);
+  const pencePart = pence.substring(pence.length - 2).padEnd(2, "0");
+  // Return formatted string
+  return `£${pounds}.${pencePart}`;
```
Comment (jaymes15) [inline Sprint-2/3-mandatory-implement/3-to-pounds.js]: 💡 Nice that you included the £ symbol directly.

**Review by jaymes15 — APPROVED** [review]
Nice One!
